core/unit_test/tools/TestWithoutInitializing.cpp

```diff
@@ -82,26 +108,46 @@ TEST(kokkosp, create_mirror_no_init_view_ctor) {
   Kokkos::View<int*, Kokkos::DefaultExecutionSpace> device_view("device view",
                                                                 10);
   Kokkos::View<int*, Kokkos::HostSpace> host_view("host view", 10);
+  auto device_memory_space =
+      typename Kokkos::DefaultExecutionSpace::memory_space{};
 
   auto success = validate_absence(
       [&]() {
         auto mirror_device = Kokkos::create_mirror(
             Kokkos::view_alloc(Kokkos::HostSpace{},
                                Kokkos::WithoutInitializing),
             device_view);
+        static_assert(
+            std::is_same_v<typename decltype(mirror_device)::memory_space,
+                           Kokkos::HostSpace>);
         auto mirror_host = Kokkos::create_mirror(
-            Kokkos::view_alloc(Kokkos::HostSpace{}, Kokkos::WithoutInitializing,
+            Kokkos::view_alloc(device_memory_space, Kokkos::WithoutInitializing,
                                Kokkos::DefaultExecutionSpace{}),
             host_view);
+    // FIXME_MSVC
+#ifndef KOKKOS_COMPILER_MSVC
+        static_assert(
+            std::is_same_v<typename decltype(mirror_host)::memory_space,
+                           decltype(device_memory_space)>);
+#endif
         auto mirror_device_view = Kokkos::create_mirror_view(
             Kokkos::view_alloc(Kokkos::HostSpace{},
                                Kokkos::WithoutInitializing),
             device_view);
+        static_assert(
+            std::is_same_v<typename decltype(mirror_device_view)::memory_space,
+                           Kokkos::HostSpace>);
```

Comment:
There are mixed usages of `std::is_same` and `std::is_same_v` throughout this PR. If you want this patch to be a candidate patch for 3.7, I don't think you can use `std::is_same_v`?

Reply:
Answering anyway: I expect that some minor adjustments would be needed for 3.7 anyway.